src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,33 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  activeTodo: Todo;
+  setActiveTodo: (currentTodo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ activeTodo, setActiveTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [userLoading, setUserLoading] = useState(false);
+
+  useEffect(() => {
+    if (activeTodo.id) {
+      setUserLoading(true);
+    }
+
+    getUser(activeTodo.userId)
+      .then(setUser)
+      .finally(() => setUserLoading(false));
+  }, [activeTodo]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {userLoading || !user ? (
```

Comment:
I think I would reverse the conditions, we should check if user is present and user is not loading, it will be more clear that way. As a developer I want to check component which should be displayed on success instead of loader

Reply:
conditions reversed